File: apple/Sources/ReactNativeWatchOS/Modifiers/Brightness.swift
import SwiftUI

/// SwiftUI `.brightness(_:)`. Adds `value` to each color component;
/// `0` leaves the view unchanged.
enum RNWBrightnessModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("brightness") { view, params, _ in
            AnyView(view.brightness(params.double("value") ?? 0))
        }
    }
}

File: src/modifiers/brightness.ts
import { createModifier } from './createModifier';

/// SwiftUI `.brightness(_:)`. Brightens (positive) or darkens (negative)
/// the view by adding `value` to each color component. `0` is unchanged.
export function brightness(value: number) {
  return createModifier('brightness', { value });
}
